MetaMask/metamask-extension
# Fix: Migrate TextField to TS

## Explanation
This PR focuses on the migration of the `TextField` component, currently in JavaScript, from the component-library folder to TypeScript. As part of this process, the JavaScript version is now deprecated. 

To minimize disruption to the application, we have updated all import paths from the JavaScript version to point to the deprecated version. 

Following the approval and merge of this PR, we will proceed to replace every instance of the deprecated JavaScript component with the new TypeScript version. We will ensure that all functionalities are working as expected after this transition.

* Fixes #19127 

## Screenshots/Screencaps

### Before


https://github.com/MetaMask/metamask-extension/assets/8112138/3e61fe03-bf41-48e5-8e9f-4618df7a5b3a



### After


https://github.com/MetaMask/metamask-extension/assets/8112138/4523091c-7826-471a-8b3f-951b991908c1



## Manual Testing Steps

- Go to the storybook build of this PR
- Search `TextField` in the search bar and go to the component-library version
- Check documentation, stories and controls 

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
